stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -109,8 +109,12 @@ julia> cross(a,b)
  0
 ```
 """
-cross(a::AbstractVector, b::AbstractVector) =
+function cross(a::AbstractVector, b::AbstractVector)
+    if !(length(a) == length(b) == 3)
+        throw(DimensionMismatch("cross product is only defined for vectors of length 3"))
+    end
     [a[2]*b[3]-a[3]*b[2], a[3]*b[1]-a[1]*b[3], a[1]*b[2]-a[2]*b[1]]
```

Comment:
Since the length is checked, this could be annotated with `@inbounds`.

Reply:
Offset vectors?

Reply:
@GregPlowman what about them?  Last I checked, `length` is not implemented for offset vectors, so this function would simply fail for them (assuming nothing has changed).

Reply:
OK, nothing to worry about then.
Sorry for the distraction.

Reply:
To be on the safe side, you can add `firstindex(a) == firstindex(b) == 1` check.
I suppose for `Vector` it should be optimized into `true` at compile time.